### PR TITLE
Adding support for Xenial

### PR DESCRIPTION
**Pull request by trumant**
Changes:
 * After successful testing with Xenial, it made sense to migrate to
distribution and version-specific vars where they differ.
 * `exec_restart_command` task and `network_interface_states_path` var
are both removed as they are not referenced anywhere
 * README and role metadata updated to call out Xenial support

### CODE REVIEW

File: vars/ubuntu-12.04.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+ubuntu-14.04.yml
```
Comment (arBmind) [inline vars/ubuntu-12.04.yml]: this looks wrong to me

File: README.md [hunk 2]
```diff
@@ -26,7 +26,6 @@ Example Playbook
 
   roles:
   - role: dresden-weekly.network-interfaces
-    network_manage_devices: yes
```
Comment (arBmind) [inline README.md:29]: why was this line removed?

**Comment by arBmind** [comment]
Thank you for the pull request. Looks very promising. 

Please check my comments on the code.

Wouldn't it help a lot if we add some default os vars? I guess most users use Ubuntu anyways but I know that Debian is also using this role.

File: README.md (hunk 2 above)
Comment (trumant) [reply README.md:29]: The var is already set to this value in the role's `defaults/main.yml`

File: vars/ubuntu-12.04.yml (hunk 1 above)
Comment (trumant) [reply vars/ubuntu-12.04.yml]: This file is just a symlink to `ubuntu-14.04` as both distro versions use the same vars. I figured it might reduce maintenance, but I'm happy for it to stand-alone as a separate file if you prefer. What is your preference?

File: README.md (hunk 2 above)
Comment (arBmind) [reply README.md:29]: Ah ok I see...

File: vars/ubuntu-12.04.yml (hunk 1 above)
Comment (arBmind) [reply vars/ubuntu-12.04.yml]: I would run into issues with symlinks on Windows. Please replicate the file.

Comment (trumant) [reply vars/ubuntu-12.04.yml]: No problem, I'll make that change.

**Comment by arBmind** [comment]
Ok, great work on this. I am happy to merge this.
